### PR TITLE
Has binding expectation

### LINKED ISSUE #41: Expectation HasBinding
Should work for procedures, functions and program.

### PR DESCRIPTION
**Pull request by faloi**
Closes #41

### CODE REVIEW

File: lib/subject_extensions.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module StonesSpec::Subject
```
Comment (faloi) [inline lib/subject_extensions.rb]: I did it this way because `ast_regexp` doesn't have sense for ruby-stones

File: lib/expectations_runner.rb [hunk 2]
```diff
@@ -19,9 +25,18 @@ class Mumukit::Inspection::PlainInspection
...
   def expectations
     {
-      'HasWhile' => /AST\(while/
+      'HasWhile' => use(/AST\(while/),
+      'HasBinding' => lambda { |binding| subject_for(binding).ast_regexp }
     }
   end
+
+  def subject_for(binding)
+    if binding == 'program'
```
Comment (faloi) [inline lib/expectations_runner.rb]: This is a bit ugly, but `Subject.from` returns `Program` only when the binding is empty. Thinking it again, I could change it to consider this case (right now it returns `Function` for input `string` and that's not right).

File: lib/expectations_runner.rb [hunk 3]
```diff
@@ -19,9 +25,18 @@ class Mumukit::Inspection::PlainInspection
 
   def expectations
     {
-      'HasWhile' => /AST\(while/
+      'HasWhile' => use(/AST\(while/),
```
Comment (faloi) [inline lib/expectations_runner.rb]: I'd love to skip the parentheses here, but I didn't know how :disappointed: 
Any ideas?

Comment (faloi) [inline lib/expectations_runner.rb]: @AlvarezAriel this is the kind of stuff you like, perhaps you have already tried it

Comment (AlvarezAriel) [inline lib/expectations_runner.rb]: Unfortunately,  you cannot avoid parentheses on function calls inside hash declarations, but you can move them like `(use /AST\(while/  )`

File: lib/subject_extensions.rb (hunk 1 above)
Comment (flbulgarelli) [inline lib/subject_extensions.rb]: It has total sense for me

File: lib/expectations_runner.rb (hunk 2 above)
Comment (flbulgarelli) [inline lib/expectations_runner.rb]: It is ok for now. Perhaps we could chang stones-spec to work on that case too

**Comment by flbulgarelli** [comment]
It is ok for me. Please procees to merge unless you want to do any change after @AlvarezAriel  comments

**Pull request merged**